src/main/java/edu/wpi/grip/ui/pipeline/AddSourceView.java

```diff
@@ -86,11 +87,55 @@ public AddSourceView(EventBus eventBus) {
             dialog.getDialogPane().setStyle(root.getStyle());
             dialog.getDialogPane().getStylesheets().addAll(root.getStylesheets());
 
-            // If the user clicks OK, add a new webcam source
+            // If the user clicks OK, add a new camera source
             dialog.showAndWait().filter(Predicate.isEqual(ButtonType.OK)).ifPresent(result -> {
                 loadSourceExecutor.execute(() -> {
                     try {
-                        final WebcamSource source = new WebcamSource(eventBus, cameraIndex.getValue());
+                        final CameraSource source = new CameraSource(eventBus, cameraIndex.getValue());
+                        eventBus.post(new SourceAddedEvent(source));
+                    } catch (IOException e) {
+                        eventBus.post(new FatalErrorEvent(e));
+                        e.printStackTrace();
```

Comment:
eventBus.post never ever ever ever throw an exception

(also this code was written by you, I just copied those lines for the IP camera source)